windows/vNext/ReactNativeVideoCPP/ReactVideoView.cpp

```diff
@@ -0,0 +1,255 @@
+#include "pch.h"
+#include "ReactVideoView.h"
+#include "ReactVideoView.g.cpp"
+#include "NativeModules.h"
+
+using namespace winrt;
+using namespace Windows::Foundation;
+using namespace Windows::Foundation::Collections;
+
+using namespace Windows::UI::Xaml;
+using namespace Windows::UI::Xaml::Media;
+using namespace Windows::UI::Xaml::Controls;
+using namespace Windows::UI::Core;
+using namespace Windows::Media::Core;
+using namespace Windows::Media::Playback;
+
+namespace winrt::ReactNativeVideoCPP::implementation {
+ReactVideoView::ReactVideoView(winrt::Microsoft::ReactNative::IReactContext const &reactContext)
+    : m_reactContext(reactContext) {
+  // always create and set the player here instead of depending on auto-create logic
+  // in the MediaPlayerElement (only when auto play is on or URI is set)

```

Comment:
Is this comment supposed to mean that there are cases when the default create logic is used? Or is it suposed to indicate that the mediaplayer is not created until the source is set?
Also I don't see autoplay handled in the updateProperties

Reply:
Inside XAML, it has logic to not create underline MediaPlayerElement object if URI is not set yet. For simplicity, I just create the object upfront so we can use it later when properties are set, without needing to cache all properties and wait for URI set.

Also in regarding to autoPlay, I am going to add logic to enable autoplay when "paused" is not set or set to false, to be consistent with other platforms.

Reply:
relevant: https://github.com/react-native-community/react-native-video/pull/1916